fuse_models/include/fuse_models/parameters/odometry_2d_params.h

```diff
@@ -77,11 +77,38 @@ struct Odometry2DParams : public ParameterBase
       nh.getParam("disable_checks", disable_checks);
       nh.getParam("queue_size", queue_size);
       getParamRequired(nh, "topic", topic);
+      getParamRequired(nh, "twist_target_frame", twist_target_frame);
+
       if (!differential)
       {
         getParamRequired(nh, "pose_target_frame", pose_target_frame);
       }
-      getParamRequired(nh, "twist_target_frame", twist_target_frame);
+      else
+      {
+        nh.getParam("independent", independent);
+
+        if (!independent)
+        {
+          std::vector<double> minimum_pose_relative_covariance_diagonal(3, 0.0);
+          nh.param("minimum_pose_relative_covariance_diagonal", minimum_pose_relative_covariance_diagonal,
+                   minimum_pose_relative_covariance_diagonal);
+
+          if (minimum_pose_relative_covariance_diagonal.size() != 3)
+          {
+            throw std::runtime_error("Minimum pose relative covariance diagonal must be of length 3!");
+          }
+
+          if (std::any_of(minimum_pose_relative_covariance_diagonal.begin(),
```

Comment:
Unrelated to this PR, but I'm guessing that we need to validate covariance diagonals in quite a few places. Might be an opportunity for a utility method somewhere.

Reply:
If you don't mind, I can do that in another PR.

I also want to move the checks for **symmetry** and **PSD-ness** to a header in `fuse_core`. That way I can also unit test it, and for the case of **symmetry** I want to stop using `isApprox`, because it doesn't play well with zero (and close to zero) matrices, and for a symmetry check I think it's more intuitive to do a simply element-wise comparison against the user-given precision. I believe all this is enough to justify another PR. I promise I'll send it during this week. :pray:

Reply:
Sorry! I should clarify: this is great as-is, and even the change I proposed was more of a musing for another time (for anyone to tackle), rather than a concrete ask.

Reply:
I've done this in https://github.com/locusrobotics/fuse/pull/139

Since the changes are on top of this PR, this should be merged before you can review https://github.com/locusrobotics/fuse/pull/139